public/js/reducers/sources.js

```diff
@@ -224,15 +215,15 @@ function getSourceText(state: OuterState, id: string) {
 }
 
 function getSourceTabs(state: OuterState) {
-  return state.sources.tabs;
+  const tabs = state.sources.tabs;
+  return tabs
+    .filter(url => getSourceByURL(state, url))
+    .map(url => getSourceByURL(state, url));
```

Comment:
Hmm, we should have something like navigate for chrome.
On Sun, Aug 28, 2016 at 6:06 PM Jaideep Bhoosreddy <notifications@github.com>
wrote:

> In public/js/reducers/sources.js
> <https://github.com/devtools-html/debugger.html/pull/603#discussion_r76540123>
> :
>
> > @@ -224,15 +215,15 @@ function getSourceText(state: OuterState, id: string) {
> >  }
> >
> >  function getSourceTabs(state: OuterState) {
> > -  return state.sources.tabs;
> > +  const tabs = state.sources.tabs;
> > +  return tabs
> > +    .filter(url => getSourceByURL(state, url))
> > +    .map(url => getSourceByURL(state, url));
>
> I took "url" from the navigate/will-navigate event payload from the
> tabTarget. But more importantly, this would only work for Firefox. I'll
> look to see if I can do something for chrome.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/devtools-html/debugger.html/pull/603/files/2879e218ebefe6b97861876a62646467a1bbbb1f#r76540123>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAPiYhNVT_zyaNxixxhD_6nzz-sLcSQCks5qkgZrgaJpZM4Juo30>
> .
>